game/scripts/vscripts/heroes/meepo/divided_we_stand.ts

```diff
@@ -0,0 +1,352 @@
+          mainMeepo.SetAbsOrigin(oldLocation)
+
+          // Make sure the hero survives this
+          parent.SetHealth(parent.GetMaxHealth())
+        }
+        for (let meepo of GetAllMeepos(mainMeepo)) {
+          if (meepo != mainMeepo) {
+            meepo.AddNewModifier(mainMeepo,this.GetAbility(),"modifier_meepo_divided_we_stand_death_lua",{})
+          }
+        }
+      }
+    }
+  }
+
+  // Death is dealt with in OnTakeDamge
+  /*OnDeath(keys) {
+    if (keys.unit != this.GetParent()) {return null}
+
+    let mainMeepo = <CDOTA_BaseNPC_Hero>this.GetCaster();
+    let parent = <CDOTA_BaseNPC_Hero> this.GetParent()
+
+    for (let meepo of GetAllMeepos(mainMeepo)) {
+      if (parent != meepo) {
+        meepo.AddNewModifier(mainMeepo,this.GetAbility(),"modifier_meepo_divided_we_stand_death_lua",{})
+      }
+    }
+  }*/
+  
+
+  OnRespawn(keys) {
+    let parent = <CDOTA_BaseNPC_Hero> this.GetParent()
+    let mainMeepo = <CDOTA_BaseNPC_Hero> this.GetCaster()
+    if (keys.unit == parent && parent == PlayerResource.GetSelectedHeroEntity(this.GetParent().GetPlayerOwnerID())) {
+      for (let meepo of GetAllMeepos(mainMeepo)) {
+        if (meepo != mainMeepo) {  
+          meepo.RemoveModifierByName("modifier_meepo_divided_we_stand_death_lua")
+          meepo.RemoveNoDraw()
+          FindClearSpaceForUnit(meepo,this.GetParent().GetAbsOrigin(),true)
+          meepo.AddNewModifier(meepo, this.GetAbility(), "modifier_phased", {duration:0.1})
+        }
+      }
+    }
+  }
+}
+// Use this to fake deaths
+class modifier_meepo_divided_we_stand_death_lua extends CDOTA_Modifier_Lua {
+  IsPermanent() { return false}
+  IsHidden() { return true }
+
+  OnCreated() {
+    if (IsServer()) {
+      this.GetParent().StartGesture(GameActivity_t.ACT_DOTA_DIE)
+      this.StartIntervalThink(1.5)
+    }
+  }
+
+  OnIntervalThink() {
+    let parent = this.GetParent()
+    parent.RemoveGesture(GameActivity_t.ACT_DOTA_DIE)
+    parent.AddNoDraw()
+    if (parent.GetTeamNumber() == DOTATeam_t.DOTA_TEAM_GOODGUYS) {
+      parent.SetAbsOrigin(Vector(-10000,-10000,0))
+    } else {
+      parent.SetAbsOrigin(Vector(10000,10000,0))
+    }
+    this.StartIntervalThink(-1) 
+  }
+
+
+  CheckState() {
+    return {
+      [modifierstate.MODIFIER_STATE_STUNNED]: true,
+      [modifierstate.MODIFIER_STATE_UNSELECTABLE]:true,
+      [modifierstate.MODIFIER_STATE_INVULNERABLE]:true,
+      [modifierstate.MODIFIER_STATE_UNTARGETABLE]:true,
+      [modifierstate.MODIFIER_STATE_OUT_OF_GAME]:true,
+      [modifierstate.MODIFIER_STATE_NO_HEALTH_BAR]:true,
+      [modifierstate.MODIFIER_STATE_NOT_ON_MINIMAP]:true,
+    }
+  }
+}
+
+// LevelAbilityForAllMeepos
+// This can't be done in the onupgrade blocks because of talents
+function LevelAbilitiesForAllMeepos(caster:CDOTA_BaseNPC_Hero): void {
+  
+  let PID = caster.GetPlayerOwnerID();
+  let mainMeepo:CDOTA_BaseNPC_Hero = PlayerResource.GetSelectedHeroEntity(PID);
```

Comment:
why only put type at one of the declarations, either put it at both or neither, PlayerResource.GetSelectedHeroEntity should return `CDOTA_BaseNPC_Hero` by default according to declarations.